ansible/roles/docker/tasks/main.yml

```diff
@@ -8,20 +8,25 @@
     gpgkey: https://download.docker.com/linux/centos/gpg
   tags: docker
 
-- name: install docker-ce-17.09.0.ce
-  yum: name=docker-ce-17.09.0.ce state=present
+- name: install docker-ce-17.09.1.ce
+  yum: name=docker-ce-17.09.1.ce state=present allow_downgrade=yes
```

Comment:
Would this work if we already have a version lock in place for the package?

Reply:
@mxtthias Well... No, that's true. It's not elegant, but I've worked around it by removing the lock before install (and possible downgrade), and adding it back afterwords, in commit 4511943.

I don't know how this will work on next update really. Docker has this annoying practice of obsoleting things aggressively and really forcing you into their preferred RPM package. See e.g. https://github.com/moby/moby/issues/33930 and https://github.com/docker/for-linux/issues/20#issuecomment-312122325

We should be able to find out if it really-really works by manually installing `docker-ce-18.01.0.ce` when it's released, then run the playbook again to verify it does indeed successfully downgrade to `17.09.1`.

OR, we dodge this mess by reverting the extra complexity added with 4511943, and simply
1. install *any* stable docker-ce version that the Docker repo prefers
2. forbid it to auto-upgrade

That's probably more in line with what Docker folks want, so it's easier. Meaning this:

```yaml

- name: install docker-ce
  yum: name=docker-ce state=present
  register: yum_install_docker
  tags: docker

- name: lock docker-ce version
  shell: yum versionlock docker-ce
  when: yum_install_docker.changed
  tags: docker

```

Instead of this!

```yaml

- name: remove docker-ce version lock
  shell: yum versionlock delete docker-ce
  register: versionlock_delete_docker
  changed_when: '"versionlock delete: no matches" in versionlock_delete_docker.stdout'
  failed_when: no
  tags: docker

- name: install docker-ce-17.09.1.ce
  yum: name=docker-ce-17.09.1.ce state=present allow_downgrade=yes
  tags: docker

- name: add docker-ce version lock
  shell: yum versionlock docker-ce
  changed_when: no
  tags: docker

```

@mxtthias What do you think?

Reply:
Not specifying a version makes me a bit uneasy. Even if we make sure we only run that task once, when we set up the machine, we'll get unpredictable results if/when we need to do it next time.

Specifying a version will require us to make sure that the version is actually available when we need to install it, which I guess means that we'll have to make sure we stay up to date with the current `docker-ce` version..

Neither way is awesome, but I lean towards keeping the version specified. What are your thoughts?

Reply:
@mxtthias 

> Neither way is awesome, but I lean towards keeping the version specified. What are your thoughts?

I agree! Then the current revision should be what we're seeking. Ready to approve?